### DISCUSSION TITLE
APIBlueprint & APIVew: differences and compatibility (migration from flask_smorest)

### DISCUSSION THREAD

**Discussion by awoimbee** [luolingchun/flask-openapi3#157] [Q&A]
Hi,

Using `flask_smorest`, I have routes defined like:
```python
blp = Blueprint(
    "objects",
    "objects",
    url_prefix="/objects",
    description="Operations on objects",
)

@blp.route("/")
class AnyObject(MethodView):
    pass
```

How would that look like using flask-openapi3 ? It's a bit lost, currently I have:
```python
blp = APIBlueprint(
    "objects",
    __name__,
    url_prefix="/objects",
    description="Operations on objects",
)
api_view = APIView()

@api_view.route("/")
class AnyObject:
    pass

blp.register_api_view(api_view) # this doesn't work
```

I din't understand the difference between APIView and APIBlueprint, nor why APIBlueprint seems to be required as a replacement for `flask.MethodView` ?

---

Environment:

- Python version: 3.12
- Flask version: 3.0.3
- flask-openapi3 version: 4.0.0rc2

**Comment by luolingchun** [comment]
@awoimbee

`APIBlueprint` directly inherits all the functions of `falsk.Blueprint`, and extends the relevant functions of openapi documentation on the basis of Blueprint.

In addition, `APIBlueprint` is used to decorate view functions, while `APIView` is used to decorate view classes, which is different.
